client_test.go

```diff
@@ -188,6 +200,38 @@ func TestClient_Get(t *testing.T) {
 	resp.Body.Close()
 }
 
+func TestClient_RequestLogHook(t *testing.T) {
+	// Mock server which always responds 500.
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "GET" {
+			t.Fatalf("bad method: %s", r.Method)
+		}
+		if r.RequestURI != "/foo/bar" {
+			t.Fatalf("bad uri: %s", r.RequestURI)
+		}
+		w.WriteHeader(200)
+	}))
+	defer ts.Close()
+
+	retries := -1
+
+	client := NewClient()
```

Comment:
Can we set a logger on the new client, write to it in the callback and check its buffer? Or at least just ensure the same logger instance is passed through.